app/routes/song/index.js

```diff
@@ -0,0 +1,19 @@
+import { Router } from 'express';
+import { SongController } from '../../controllers';
+import { AuthMiddleware, ArtistMiddleware, UploadMiddleware } from '../../middlewares';
+
+const { authenticate } = AuthMiddleware;
+const { createSong } = SongController;
+const { artistValidator } = ArtistMiddleware;
+const { fileUpload } = UploadMiddleware;
+const router = Router();
+
+router.post(
+  '/:albumId || null',
```

Comment:
- If you will have a param on ``/song`` it should be ``:songId``, hence album id should probably come with the body in a case like this.
- You don't have to the pipe to check if null, if not sent, it is null by default